ext/reflection/php_reflection.c

```diff
@@ -2259,6 +2312,242 @@ ZEND_METHOD(reflection_parameter, __construct)
 			/* returns out of this function */
 	}
 
+	*func_ptr = fptr;
+	*out_ce = ce;
+}
+
+/* {{{ proto public ReflectionReturnType ReflectionFunction::getReturnType()
+   Returns a return type object for this function, or FALSE if no return type */
+ZEND_METHOD(reflection_function, getReturnType)
+{
+	reflection_object *intern;
+	return_type_reference *reference;
+	zend_function *fptr;
+	struct _zend_arg_info *arg_info;
+
+	METHOD_NOTSTATIC(reflection_function_abstract_ptr);
+	GET_REFLECTION_OBJECT_PTR(fptr);
+
+	if (!fptr->common.arg_info) {
+		RETURN_FALSE;
+	}
+
+	arg_info = &fptr->common.arg_info[-1];
+
+	if (!arg_info->type_hint) {
+		RETURN_FALSE;
+	}
+
+	reference = (return_type_reference *) emalloc(sizeof(return_type_reference));
+	reference->arg_info = arg_info;
+	reference->fptr = fptr;
+
+	reflection_instantiate(reflection_return_type_ptr, return_value);
+	Z_REFLECTION_P(return_value)->ptr = reference;
+	Z_REFLECTION_P(return_value)->ref_type = REF_TYPE_RETURN_TYPE;
+}
+/* }}} */
+
+/* {{{ proto public ReflectionReturnType::__construct(mixed function)
+   */
+ZEND_METHOD(reflection_return_type, __construct)
+{
+	zval *reference = NULL;
```

Comment:
since reference is a required parameter,  initialized it to NULL make no sense at all.